### PR TITLE
Add ARM support

### LINKED ISSUE #125: Add Support for A1 Instance type
**What would you like to be added**:
EKS Optimized AMI for A1 Instance 
**Why is this needed**:

- Explore possibilities to optimize cost/performance

### PR DESCRIPTION
**Pull request by mcrute**
*Issue #:* Fixes #125

*Description of changes:*
Adds support for the ARM a1 instance class to the AMI builder.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: files/eni-max-pods.txt [hunk 1]
```diff
@@ -11,6 +11,11 @@
 # access the instance metadata, VPC DNS, and Time Sync services from the 32nd IP
 # address onwards. If access to these services is needed from all IP addresses
 # on the interface, we recommend using a maximum of 31 IP addresses per interface.
+a1.medium 8
+a1.large 14
+a1.xlarge 20
+a1.2xlarge 20
+a1.4xlarge 39
```
Comment (mogren) [inline files/eni-max-pods.txt]: Hmm, I think this should be:
```
a1.medium 8
a1.large 29
a1.xlarge 58
a1.2xlarge 58
a1.4xlarge 234
```
See https://github.com/awslabs/amazon-eks-ami/pull/236/files#diff-6ec08afbedd68be4a24ff9c1f9ca668fR14

File: Makefile [hunk 2]
```diff
@@ -1,48 +1,71 @@
...
 validate:
-	packer validate eks-worker-al2.json
+	packer validate \
+		-var instance_type=$(INSTANCE_TYPE) \
+		eks-worker-al2.json
 
-1.10: validate
+.PHONY: k8s
+k8s: validate
+	@echo "$(T_GREEN)Building AMI for version $(T_YELLOW)$(VERSION)$(T_GREEN) on $(T_YELLOW)$(ARCH)$(T_RESET)"
```
Comment (mogren) [inline Makefile]: Fancy with all the colors. 👍

File: Makefile [hunk 3]
```diff
@@ -1,48 +1,71 @@
...
+	$(eval SOURCE_AMI_ID := $(shell aws ec2 describe-images \
+		--output text \
+		--filters \
+			Name=owner-id,Values=137112412989 \
```
Comment (mogren) [inline Makefile]: `SOURCE_AMI_OWNERS=137112412989` at the top, then `$(SOURCE_AMI_OWNERS)` here?

**Comment by left4taco** [comment]
@mcrute Hi there. I'm trying to use your script to create Arm AMI. But it looks like aws-cli isn't able to find the base image. 
```bash
$ aws ec2 describe-images \
                --output text \
                --filters \
                        Name=owner-id,Values=137112412989 \
                        Name=virtualization-type,Values=hvm \
                        Name=root-device-type,Values=ebs \
                        Name=name,Values=amzn2-ami-minimal-hvm-* \
                        Name=architecture,Values=arm64 \
                        Name=state,Values=available \
                --query 'max_by(Images[], &CreationDate).ImageId'
None
```

This is just the first step of building arm image. Am I using it wrong? Thanks.

**Comment by mcrute** [comment]
@left4taco what region are you trying to use for AMI building? A1 instances are not [available in all regions](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services/) yet and you won't be able to build the AMI in an unsupported region.

**Comment by left4taco** [comment]
@mcrute I'm using us-west-2 Oregon. A1 instances are available in this region. 

It turned out that `aws ec2 describe-images` command in `Makefile` is using the default one of the system, instead of the one specified in `eks-worker-al2.json`. Problem solved. Thanks.

File: Makefile [hunk 4]
```diff
@@ -1,48 +1,72 @@
...
-	packer validate eks-worker-al2.json
+	packer validate \
+		-var instance_type=$(INSTANCE_TYPE) \
+		eks-worker-al2.json
 
-1.10: validate
+.PHONY: k8s
+k8s: validate
+	@echo "$(T_GREEN)Building AMI for version $(T_YELLOW)$(VERSION)$(T_GREEN) on $(T_YELLOW)$(ARCH)$(T_RESET)"
+	$(eval SOURCE_AMI_ID := $(shell aws ec2 describe-images \
```
Comment (sinneduy) [inline Makefile]: @mcrute I don't follow why this is necessary.  Shouldn't packer's `source_ami_filter` be able to choose based on the architecture passed in [here](https://github.com/awslabs/amazon-eks-ami/blob/master/eks-worker-al2.json#L34)?

Doesn't doing this force the `Makefile` to always force using the `source_ami` instead?  https://github.com/awslabs/amazon-eks-ami/blob/master/eks-worker-al2.json#L30

Could you provide some additional insight here?

**Pull request merged**